src/actions/logActions.ts

```diff
@@ -51,6 +51,44 @@ export const deleteLogDialogThunkAsync = (app: AppBase, logDialogId: string, pac
     }
 }
 
+const deleteLogDialogsAsync = (appId: string): ActionObject => {
+    return {
+        type: AT.DELETE_LOG_DIALOGS_ASYNC,
+        appId,
+    }
+}
+
+const deleteLogDialogsFulfilled = (logDialogIds: string[]): ActionObject => {
+    return {
+        type: AT.DELETE_LOG_DIALOGS_FULFILLED,
+        logDialogIds,
+    }
+}
+
+const deleteLogDialogsRejected = (): ActionObject => {
+    return {
+        type: AT.DELETE_LOG_DIALOGS_REJECTED,
+    }
+}
+
+export const deleteLogDialogsThunkAsync = (app: AppBase, logDialogIds: string[], packageId: string) => {
+    return async (dispatch: Dispatch<any>) => {
+        dispatch(deleteLogDialogsAsync(app.appId))
+        const clClient = ClientFactory.getInstance(AT.DELETE_LOG_DIALOGS_ASYNC)
+
+        try {
+            await clClient.logDialogsDeleteMany(app.appId, logDialogIds)
```

Comment:
Since we don't block the UI, is there a point where user can open one of the log dialogs before the deleted is completed and end up in a weird state?

Reply:
Confirm this was possible.  I did what we talked about in person, moving the removal of the dialogs from the store so the `ASYNC` call instead of `FULFILLED` call so they happen preemptively.